java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java

```diff
@@ -52,18 +53,8 @@ public ApproxEqualsVisitor(ValueVector left, ValueVector right, float floatEpsil
     this.floatEpsilon = floatEpsilon;
     this.doubleEpsilon = doubleEpsilon;
 
-    floatDiffFunction =
-        (float value1, float value2) -> Math.abs(value1 - value2) <= floatEpsilon;
-    doubleDiffFunction =
-        (double value1, double value2) -> Math.abs(value1 - value2) <= doubleEpsilon;
-  }
-
-  public void setFloatDiffFunction(DiffFunctionFloat floatDiffFunction) {
-    this.floatDiffFunction = floatDiffFunction;
-  }
-
-  public void setDoubleDiffFunction(DiffFunctionDouble doubleDiffFunction) {
```

Comment:
We shouldn't limit diffFunction to Math.abs(d1 - d2) <= epsilon(i.e. Validator.java has different implementation)? To make it more flexible, we should keep APIs like these two for setting special diffFunction?

Reply:
Sure. Agreed with you. Thanks. 
I have updated the code to allow arbitrary equalization functions. In addition, we have provided copy operation to the equalization function to avoid problem related to thread-safe.
Please take a look.